Add a diagram of key pieces to quick start, tweak some wording. (#184)
dbos-inc/dbos-docs: docs/getting-started/quickstart.md

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import useBaseUrl from '@docusaurus/useBaseUrl';
import ThemedImage from '@theme/ThemedImage';

# DBOS Quickstart

> Let's create your first DBOS app! You start building an app by creating a special folder for it. From there, you can register with DBOS Cloud, provision a free Database Instance and deploy your app in a few minutes. You can also run DBOS apps on your computer.

Reviewer: This still needs improvement. The paragraph is so dumbed down that it's confusing for someone who knows how to program. For example, the "special folder" isn't special in any way--it's just a regular folder with code in it. I still prefer my version:  "If you build an app with DBOS, you can develop it on your computer, then deploy it in seconds to DBOS Cloud," but I know it can also be improved.

Author: Should be all set now

<ThemedImage
  alt="Docusaurus themed image"
  sources={{
    light: useBaseUrl('/img/quickstart-diagram.png'),

>     dark:  useBaseUrl('/img/quickstart-diagram-dark.png'),

Reviewer: Dark mode image is nice, but still doesn't look great--the contrast between, for example, the app folder and the text on it is low.

Author: Pushed another version in

  }}
/>

In this guide, we start by deploying a sample "Hello" app to the cloud. After that, we provide instructions for how to run this app on your system.

## Deploy your First App to the Cloud

> ### 1. Install Node.js version 20 or later

Reviewer: Can we upgrade these instructions to node 22 (we require 20, but should be providing instructions for the latest stable version)

Author: Should be in place now

# DBOS Quickstart

> You start developing a DBOS app by creating a folder for it on your computer. From this folder, you can run the app locally and deploy to DBOS Cloud in minutes.

Reviewer: We need to iterate a lot on this opening wording, I don't think it's great now. Here's my suggestion, which still isn't amazing:

> Let's build our first DBOS app!  If you build an app with DBOS, you can run it locally connected to your local Postgres database, or deploy it to the cloud connected to a cloud Postgres database. In this guide, we'll show you how to do both.

Author: I know what you mean. I put like 2 hours into that sentence already. And no I don't like yours as much yet :) I think this beginning should be simple....

I'll think about it some more.

Author: Should be all set now

> ![DBOS Development and Deployment Model](./assets/quickstart-diagram.png)

Reviewer: The image looks great in light mode but terrible in dark mode. How hard is it to create a dark mode version of it?

Author: will do

Author: Done

npx dbos-cloud app deploy
```

> This command first prompts you to login, or register if this is your first time. Then, it prompts you to provision a free cloud database instance. Finally, it uploads your code to DBOS Cloud and deploys your app. After about a minute, it should succeed and print `Successfully deployed <app-name>! Access your application at <URL>`.

Reviewer: ```suggestion
This command first prompts you to login, or register if this is your first time. Then, it prompts you to provision a free cloud database instance. Finally, it uploads your code to DBOS Cloud and deploys your app. In less than a minute, it should succeed and print `Successfully deployed <app-name>! Access your application at <URL>`.
```

Author: Unlike promotional materials, I think doc can set medium expectations and the software can then pleasantly exceed them? See what you think about my latest?

Author: Should be all set now

To see that your app is working, visit `<URL>` in your browser. For example, if your username is `mike` and your app name is `hello`, you would visit:
```
https://mike-hello.cloud.dbos.dev/
```

Congratulations, you've successfully deployed your first app to DBOS Cloud! You can see your deployed app in the [cloud console](https://console.dbos.dev/), or in the CLI by running `npx dbos-cloud app list`.

## Run the App on Your Computer

For development, testing, or self-hosted deployment, here's how to run this app on your local machine. This section assumes you've already created an app folder as described above.

> ### 1. Install Postgres or use Docker

Reviewer: To make things even simpler, let's combine steps 1 and 2 here. Their being separate makes the doc look complicated.

Author: Done

npx dbos-cloud app deploy
```

> This command first prompts you to login, or register if this is your first time. Then, it prompts you to provision a database instance. Finally, it uploads your code to DBOS Cloud and deploys your app. In about a minute, it should succeed and print `Successfully deployed <app-name>! Access your application at <URL>`.

Reviewer: It really is just 20 seconds now--don't think we need to say a minute.

Author: Should be all set now

   ```
  </TabItem>
    <TabItem value="linux" label="Linux">

>    Follow the [Docker Engine installation page](https://docs.docker.com/engine/install/) to install Docker on several popular Linux distributions.

Reviewer: I really like how those steps are now combined! We can drop the :::tip I think.

Author: I would like to defend that tip (with my honor). I put that there last PR because I really needed it when it was my time. It would've saved me like an hour.

It was something like "How am I supposed to know there's a record in the database? Let me connect and look at the schema. Okay but how do I connect to this docker thing? How am I supposed to know that the port 5432 is forwarded to the outside?" I suppose I could've inferred it from knowing that the app talks to it but... it took a while :)
